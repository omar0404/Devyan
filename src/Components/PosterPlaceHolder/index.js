import PosterPlaceHodler from './PosterPlaceHolder';
export default PosterPlaceHodler;
