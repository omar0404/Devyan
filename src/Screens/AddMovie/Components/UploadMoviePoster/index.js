import UploadMoviePoster from './UploadMoviePoster';
export default UploadMoviePoster;
