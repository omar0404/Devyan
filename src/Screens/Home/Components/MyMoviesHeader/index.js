import MyMoviesHeader from './MyMoviesHeader';
export default MyMoviesHeader;
