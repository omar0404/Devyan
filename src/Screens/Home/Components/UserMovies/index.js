import UserMovies from './UserMovies';
export default UserMovies;
