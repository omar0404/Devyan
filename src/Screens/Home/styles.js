import {StyleSheet} from 'react-native';
import theme from 'Devyan/src/Theme';
const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 15,
    backgroundColor: theme.backgroundColor,
  },
});
export default styles;
