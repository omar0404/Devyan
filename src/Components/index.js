import Section from './Section';
import PosterPlaceHolder from './PosterPlaceHolder';
import MovieCard from './MovieCard';
export {MovieCard, Section, PosterPlaceHolder};
