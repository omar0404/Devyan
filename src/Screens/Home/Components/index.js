import MyMovies from './MyMovies';
import AllMovies from './AllMovies';
import UserMovies from './UserMovies';
import MyMoviesHeader from './MyMoviesHeader';
export {MyMovies, AllMovies, UserMovies, MyMoviesHeader};
