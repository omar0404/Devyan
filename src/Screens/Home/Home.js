import React, {useState} from 'react';
import {FlatList} from 'react-native';
import {MyMovies, AllMovies} from './Components';
import styles from './styles';
import {useNavigation} from '@react-navigation/native';
const Home = () => {
  const navigation = useNavigation();
  const [likedMovies, setLikedMovies] = useState({});

  const handleUpdateLikedMovies = (movie) => {
    const movieId = movie.id;
    if (likedMovies[movieId]) {
      delete likedMovies[movieId];
      setLikedMovies((prevLikedMovies) => ({...prevLikedMovies}));
      return;
    }
    setLikedMovies((prevLikedMovies) => ({
      ...prevLikedMovies,
      [movieId]: movie,
    }));
  };
  const handleMoviePress = (movie) => {
    navigation.navigate('Movie', {movie});
  };
  return (
    <>
      <FlatList
        style={styles.container}
        ListHeaderComponent={
          <MyMovies
            onMoviePress={handleMoviePress}
            movies={likedMovies}
            onUnLikePress={handleUpdateLikedMovies}
          />
        }
        ListFooterComponent={
          <AllMovies
            onMoviePress={handleMoviePress}
            likedMovies={likedMovies}
            onLikePress={handleUpdateLikedMovies}
          />
        }
      />
    </>
  );
};
export default Home;
