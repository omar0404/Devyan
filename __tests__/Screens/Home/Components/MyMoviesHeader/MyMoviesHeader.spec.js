import React from 'react';
import {MyMoviesHeader} from 'Devyan/src/Screens/Home/Components';

import {create} from 'react-test-renderer';

describe('MyMoviesHeader Component', () => {
  it('should render correctly', () => {
    let wrapper = create(<MyMoviesHeader />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
