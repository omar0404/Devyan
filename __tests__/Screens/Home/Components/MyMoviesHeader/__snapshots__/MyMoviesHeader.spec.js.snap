// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyMoviesHeader Component should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": "auto",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#991F25",
          "height": 15,
          "width": 5,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "rgb(255, 255, 255)",
          "fontSize": 18,
          "marginLeft": 8,
        }
      }
    >
      My Movies
    </Text>
  </View>
</View>
`;
